src/components/GradeBookHeader.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Course } from 'api';
+import TextInput from './TextInput';
+import HeaderGradeDisplay from './HeaderGradeDisplay';
+export interface GradeBookHeaderInterface {
+	currentCourse: Course;
+	updateCurrentCourse: (course: Course) => void;
+}
+const Body = styled.div`
+	display: grid;
+	grid-template-columns: 5fr 1fr 5fr 5fr;
+	grid-template-areas: 'current
+		.
+		needed
+		desired';
+`;
+const Header = styled.div`
+	color: ${({ theme }) => theme.colors.text[1]};
+	padding: 4em 3em 3em 3em;
+`;
+const CourseTitle = styled.h1`
+	font-size: ${({ theme }) => theme.fontSizes.xl};
+`;
+
+const CurrentGrade = styled(HeaderGradeDisplay)`
+	grid-area: current;
+	padding-bottom: 1em;
+	border-right: 1px solid ${({ theme }) => theme.colors.gray[1]};
+`;
+const NeededGrade = styled(HeaderGradeDisplay)`
+	grid-area: needed;
+`;
+const DesiredGrade = styled(TextInput)`
+	grid-area: desired;
+	padding-bottom: 1em;
+`;
+function GradeBookHeader({
+	currentCourse,
+	updateCurrentCourse,
+}: GradeBookHeaderInterface) {
+	const { name, desiredGrade } = currentCourse;
+	const [tempDesiredGrade, setTempDesiredGrade] = useState(
+		desiredGrade?.toString() || ''
+	);
+	function handleChange(change: Partial<Course>) {
+		const newCourse = { ...currentCourse, ...change };
+		updateCurrentCourse(newCourse);
```

Comment:
Just curious on what is `Partial<Course>` and what the function does if you could explain a little bit!

Reply:
It allows for only some of the interface of the generic provided to be used :) It is being used here to change specific attributes of the course provided 
Thank you for reviewing my 2 tickets! I appreciate it